src/pyggdrasil/tree_inference/_mcmc_sampler.py

```diff
@@ -0,0 +1,164 @@
+"""Sampler for Markov Chain Monte Carlo inference for mutation trees
+    according to the SCITE model.
+
+Note:
+    This implementation assumes that the false positive
+    and false negative rates are known and provided as input.
+"""
+
+import jax
+from typing import Tuple
+from pathlib import Path
+import logging
+
+from jax import Array
+
+
+import pyggdrasil.tree_inference._mcmc as mcmc
+import pyggdrasil.tree_inference._logprob as logprob
+import pyggdrasil.tree_inference._mcmc_util as mcmc_util
+import pyggdrasil.serialize as serialize
+
+from pyggdrasil.tree_inference._mcmc import MoveProbabilities
+from pyggdrasil.tree_inference._tree import Tree
+from pyggdrasil.tree_inference._interface import (
+    MutationMatrix,
+    JAXRandomKey,
+    ErrorRates,
+)
+
+
+def mcmc_sampler(
+    rng_key: JAXRandomKey,
+    init_tree: Tree,
+    error_rates: ErrorRates,
+    move_probs: MoveProbabilities,
+    data: MutationMatrix,
+    num_samples: int,
+    output_dir: Path,
+    num_burn_in: int = 0,
+    thinning: int = 0,
+    iteration: int = 0,
+    **kwargs,
```

Comment:
How much are `**kwargs` used below? From a quick look I think `timestamp: Optional[str] = None` would be a better choice here. (I can also be wrong.)

Reply:
Oh perfect. This solves my issue. I only used `**kwargs` to pass a single argument optionally without `pyright` complaining. - This is BETTER !